examples/common/utility/utility.hpp

```diff
@@ -356,6 +358,46 @@ class cli_argument_pack {
     }
 }; // class cli_argument_pack
 
+// utility class to aid relative error measurement of samples
+class measurements {
+public:
+    measurements() {
+        clear();
+    }
+    void clear() {
+        _secPerFrame.clear();
+    }
+    void start() {
+        _startTime = std::chrono::steady_clock::now();
+    }
+    void stop() {
+        _endTime = std::chrono::steady_clock::now();
+        std::chrono::duration<double> duration = _endTime - _startTime;
+        // store the duration in seconds
+        _secPerFrame.push_back(duration.count());
```

Comment:
Usually, it makes sense to introduce as little as possible overhead during the measurements to avoid influence the the computations being measured. So, consider pushing the postprocessing step to a later stage. Here only store `_startTime` and `_endTime`, and post process them in the independent from the main computations part, e.g., `computeRelError`.

Reply:
Thanks, Aleksei, for bringing this point. I wasn't sure which approach is less intrusive. 
Currently I have only one store with a subtract instruction. In the approach you suggested we need to 2 vector stores in the measurement loop. I was under the impression subtract + 1 store is less intrusive, please correct me if I am wrong.

Reply:
In any case I think I should reserve some memory for the vector stores based on the iteration count. I will update the constructor accordingly.

Reply:
That's correct! We want to have the impact from the measurement system itself as little as possible. Therefore, we need to consider implementing at least the following:
* Having the memory pre-allocated. I guess just need to add a parameter to the constructor.
  * Are there cases when the number of iterations is not known beforehand?
* Do all the calculations only after the measurement loop is completed. This means inside the measurement loop can only be stores of the retrieved timings.

Reply:
Modified the constructor to accept a parameter for number of iterations and reserve the memory accordingly. All calculations are done after the measurement loop is completed.